pkg/cmd/get.go

```diff
@@ -142,9 +235,144 @@ func (o *GetOptions) Complete(args []string) error {
 		o.APIPath = fmt.Sprintf("%s?labelSelector=%s", o.APIPath, url.QueryEscape(o.LabelSelector))
 	}
 
+	// Pre-compute KubeArchive query parameters
+	o.kubearchiveQueryParams = o.buildKubeArchiveQueryParams()
+
 	return nil
 }
 
+// processK8sResourcesOptimized processes Kubernetes resources with optimized sorting and filtering
+func (o *GetOptions) processK8sResourcesOptimized(resources []*unstructured.Unstructured) ([]*ResourceWithAvailability, bool) {
+	result := make([]*ResourceWithAvailability, 0, min(len(resources), o.Limit))
+	var k8sTrimmed bool
+	for _, resource := range resources {
+		if !o.passesTimestampFilter(resource) {
+			continue
+		}
+
+		resourceWithAvailability := &ResourceWithAvailability{
+			Resource:  resource,
+			InCluster: true,
+			Archived:  false,
+		}
+
+		// Insert in sorted order (newest first) using binary search
+		idx := o.findInsertionIndex(result, resource)
+
+		// Insert at the correct position
+		if idx < len(result) {
+			result = append(result, nil) // Add space
+			copy(result[idx+1:], result[idx:])
+			result[idx] = resourceWithAvailability
+		} else {
+			result = append(result, resourceWithAvailability)
+		}
+
+		// Trim to limit to save memory
+		if len(result) > o.Limit {
+			result = result[:o.Limit]
+			k8sTrimmed = true
+		}
```

Comment:
This seems inefficient to keep appending a single item to the list and then trimming it. I think it makes more sense to just declare the slice as `min(len(resources), o.Limit)+1` and just trim before returning.

An example of this was implemented for `keepLastWhen` in [this method](https://github.com/kubearchive/kubearchive/blob/main/cmd/vacuum/publisher.go#L108).